atr/get/sbom.py

```diff
@@ -378,45 +388,96 @@ def _missing_tally(items: list[sbom.models.conformance.Missing]) -> list[tuple[s
     )
 
 
+# TODO: Update this to return either a block or something we can use later in a block for styling reasons
 def _license_tally(
     items: list[sbom.models.licenses.Issue],
-) -> list[tuple[sbom.models.licenses.Category, int, list[str | None]]]:
+    old_issues: list[sbom.models.licenses.Issue],
+) -> list[tuple[sbom.models.licenses.Category, int, int, list[str | None]]]:
     counts: dict[sbom.models.licenses.Category, int] = {}
     components: dict[sbom.models.licenses.Category, list[str | None]] = {}
+    new_count = 0
+    old_map = {lic.component_name: (lic.license_expression, lic.category) for lic in old_issues}
     for item in items:
         key = item.category
         counts[key] = counts.get(key, 0) + 1
+        name = str(item).capitalize()
+        if item.component_name not in old_map:
+            new_count = new_count + 1
+            name = f"{name} (new)"
+        elif item.license_expression != old_map[item.component_name][0]:
+            new_count = new_count + 1
+            name = f"{name} (previously {old_map[item.component_name][0]} - Category {
+                str(old_map[item.component_name][1]).upper()
+            })"
         if key not in components:
-            components[key] = [str(item)]
+            components[key] = [name]
         else:
-            components[key].append(str(item))
+            components[key].append(name)
     return sorted(
-        [(category, count, components.get(category, [])) for category, count in counts.items()],
+        [(category, count, new_count, components.get(category, [])) for category, count in counts.items()],
```

Comment:
The `new_count` is the count of overall new licenses, by the looks, but these returned rows are per category. Shouldn't the new count therefore be tracked per category rather than overall, and be custom for each category row?

Reply:
It's slightly odd (and could use some improvement I'm sure) - but this function is actually called once for each category, so although this would return the overall count if you did pass licenses with multiple categories - because warnings are all category B and errors are all category X, it does render the correct information